godot-macros/src/docs.rs

```diff
@@ -122,6 +123,26 @@ fn siphon_docs_from_attributes(doc: &[Attribute]) -> impl Iterator<Item = String
         })
 }
 
+fn xml_escape(value: &str) -> String {
```

Comment:
It seems like you often pass values that could also be moved, so is it possible to take this by value (`String`)? If yes, you could allocate the result string lazily (only if there is a character to escape) and otherwise return the parameter 1:1. Not sure if that speeds anything up though...

Reply:
Maybe you're right, but I don't feel smart enough to implement this idea properly

Tried to naively add `if !value.contains(&['&', '<', '>', '"', '\'']) { return value; }` and got mixed results: when data doesn't contain special characters, it's the fastest, but when data *contains* special characters, it's the slowest

Reply:
A less naive implementation seems a bit faster, but depending on data it's sometimes even slower than the naive implementation

```rust
let Some(start) = value.find(&['&', '<', '>', '"', '\'']) else { return value; };
let mut result = String::with_capacity(value.len());
let (a, b) = value.split_at(start);
result.push_str(a);
for c in b.chars() ...
```

Assuming that most strings are short identifiers (`f32`, `Vector3`, `RefCounted`, `init` etc.), the naive implementation seems to be the fastest

<sub>(I'm starting to regret I ever decided to bench this...)</sub>

Reply:
Since the substitution is statically defined, can't the substituted string becomes a `&'static str`? 

It's signature should look like this:
```rust
fn xml_escape(value: &str) -> &str
```

Reply:
`contains()` iterates the whole sequence twice, which may or may not be worth it?

I was thinking of something like this:
```rs
let mut replacement = None;

for (i, c) in b.chars().enumerate() {
     if /* needs replacement */ {
         if replacement.is_none() {
              // lazy init
              replacement = Some(String::from_iter(b.chars().take(i)));
         }
        // work with replacement.unwrap() as usual
     }
}
```

But you're right, it might not be worth it. Otherwise it's also fine to keep this simple, we can always add optimizations in the future :slightly_smiling_face:

Reply:
This seems to be slower when data contains special characters, I'm not sure why, but I guess loops don't like complicated logic

I don't have real data to bench, but in my test project, ~80% of strings are short identifiers, so I think I'll just add the naive implementation because it works best in this case, but this is subject to change in the future

Reply:
Makes sense, thanks!